backend/src/main/kotlin/com/ci5644/trade/controllers/api/UserController.kt

```diff
@@ -0,0 +1,72 @@
+package com.ci5644.trade.controllers.api
+
+import com.ci5644.trade.services.user.UserService
+import com.ci5644.trade.dto.UserDto
+import com.ci5644.trade.dto.UserDetailsDto
+import com.ci5644.trade.config.SecurityConstants
+import com.ci5644.trade.services.auth.AuthorizationService
+import com.ci5644.trade.config.JWT.JWTSecurityUtils
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+
+/**
+ * Controller class to handle user-related HTTP requests.
+ * Uses JWT authentication for requests and CORS configuration.
+ */
+@RestController
+@RequestMapping("/api/user")
+@CrossOrigin(origins = ["http://localhost:3000"], allowCredentials = "true")
+class UserController(private val authorizationService: AuthorizationService) {
+
+    @Autowired
+    private lateinit var userService: UserService
+
+    /**
+     * Retrieves user details based on the provided authentication token.
+     * @param authCookie String - The authentication token obtained from the cookie.
+     * @return ResponseEntity<UserDto> - ResponseEntity containing user details or an error response.
+     */
+    @GetMapping("/detail")
+    fun getUser(@CookieValue(name = SecurityConstants.AUTH_COOKIE_NAME) authCookie: String): ResponseEntity<*> {
+        return try {
+            val username = JWTSecurityUtils.getAuthUserFromJWT(authCookie);
```

Comment:
for example it would be reusable here.

But even reusing it should be logic inside a service